Geometry_Engine/Compute/SubMesh.cs

```diff
@@ -0,0 +1,108 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Create a mesh from selected faces of an existing mesh.")]
+        [Input("mesh", "Mesh to get a sub-section from.")]
+        [Input("faces", "The faces of the old mesh to carry over to the new mesh.")]
+        [Output("mesh", "Mesh composed of the faces and the vertices in those faces.")]
+        public static Mesh SubMesh(this Mesh mesh, List<Face> faces)
+        {
+            List<int> dummyList = new List<int>();
+            return SubMesh<int>(mesh, faces, ref dummyList);
+        }
+
+        /***************************************************/
+
+        [Description("Create a mesh from selected faces of an existing mesh.")]
+        [Input("mesh", "Mesh to get a sub-section from.")]
+        [Input("faces", "The faces of the old mesh to carry over to the new mesh.")]
+        [Input("vertexRelatedData", "A list where each item is related to the vertex at the same index in the mesh. \n" +
+                                    "Will be changed to relate to the new mesh's vertex list.")]
+        [Output("mesh", "Mesh composed of the faces and the vertices in those faces.")]
+        public static Mesh SubMesh<T>(this Mesh mesh, List<Face> faces, ref List<T> vertexRelatedData)
```

Comment:
Wouldn't it be possible to avoid `ref` argument here? Besides making @FraserGreenroyd unhappy, I would expect it being potentially harmful when accessed from the UI level.

I would either try to split the method into two if possible without massive overhead, or, alternatively, make it `private`.

Reply:
You're right, it became less strange than I imagined